src/write.jl

```diff
@@ -59,7 +59,7 @@ function bufferedescape(io, delim, oq, cq, e)
     n == 0 && return
     needtoescape, needtoquote = check(n, delim, oq, cq)
     seekstart(VALUE_BUFFER)
-    if needtoquote
+    if (needtoquote | needtoescape)
```

Comment:
You know parentheses aren't needed, right? :-)

Also, I think it's more standard to write `||`, since `|` is the bitwise or (even if in practice it's identical for Booleans).